src/SonarScanner.MSBuild.TFS/CoverageReportProcessorBase.cs

```diff
@@ -64,24 +64,38 @@ public bool ProcessCoverageReports()
 
             Debug.Assert(this.config != null, "Expecting the config to not be null. Did you call Initialize() ?");
 
-            // Fetch all of the report URLs
-            Logger.LogInfo(Resources.PROC_DIAG_FetchingCoverageReportInfoFromServer);
-
-            if (TryGetTrxFiles(this.config, this.settings, out var trxPaths) &&
-                trxPaths.Any() &&
-                config.GetSettingOrDefault(SonarProperties.VsTestReportsPaths, true, null) == null)
+            if (config.GetSettingOrDefault(SonarProperties.VsTestReportsPaths, true, null) != null)
+            {
+                Logger.LogInfo(Resources.TRX_DIAG_SkippingCoverageCheckPropertyProvided);
+            }
+            else
             {
-                this.config.LocalSettings.Add(new Property { Id = SonarProperties.VsTestReportsPaths, Value = string.Join(",", trxPaths) });
+                // Fetch all of the report URLs
+                Logger.LogInfo(Resources.PROC_DIAG_FetchingCoverageReportInfoFromServer);
+
+                if (TryGetTrxFiles(this.config, this.settings, out var trxPaths) &&
+                    trxPaths.Any())
+                {
+                    this.config.LocalSettings.Add(new Property { Id = SonarProperties.VsTestReportsPaths, Value = string.Join(",", trxPaths) });
+                }
             }
 
             var success = TryGetVsCoverageFiles(this.config, this.settings, out var vscoveragePaths);
-            if (success &&
-                vscoveragePaths.Any() &&
-                TryConvertCoverageReports(vscoveragePaths, out var coverageReportPaths) &&
-                coverageReportPaths.Any() &&
-                config.GetSettingOrDefault(SonarProperties.VsCoverageXmlReportsPaths, true, null) == null)
+
+            if (success && vscoveragePaths.Any())
```

Comment:
Minor - consistency: this check could be in exactly the same form as in lines 76 and 77. There's no need to store `success`.

Reply:
I'm finally not sure about this one, success is what it is returned at the end of the function (as only if the TryGetVsCoverageFiles has succeeded or not).Then as long as the if statement was split, i can't find another way to achieve this.

Reply:
As discussed offline, it seems odd that we are looking for coverage files in line 83 before we check whether the user has supplied the vsCoverageXmlReportsPaths in line 87.
I think we should check vsCoverageXmlReportsPaths first, and if it's been set return true to indicate the processing should continue.

Reply:
So that was corrected in my last commit